docs/design/yarn-v1.md

```diff
@@ -0,0 +1,169 @@
+
+Non-registry dependencies are packed and uploaded to a hosted Nexus repository. The versions of these non-registry dependencies are
+updated in package.json and yarn.lock to match those in the hosted repository.
+
+## Proposed Cachi2 Implementation Overview
+
+### Specifying a Yarn Package in a Cachi2 Request
+Cachi2 users will specify ```yarn``` as the package manager or package type for all versions of Yarn. The presence of the ```__metadata``` field in yarn.lock
+[can be used to differentiate](https://github.com/yarnpkg/berry/blob/13d5b3041794c33171808fdce635461ff4ab5c4e/packages/yarnpkg-core/sources/Project.ts#L374)
+between Yarn v1.x and Yarn v2+ and route the request to the correct Yarn module in Cachi2.
+
+### Offline Mirror
+Prefetching dependencies for Yarn in Cachi2 will be done using Yarn's [offline mirror](https://classic.yarnpkg.com/blog/2016/11/24/offline-mirror/)
+feature. When a project is configured to use the offline mirror, Yarn will store compressed archives in the mirror directory and can install
+cached project dependencies from there later without network access.
+
+The offline mirror also has a setting that controls [whether it removes package archives](https://classic.yarnpkg.com/en/docs/prune-offline-mirror/)
+from the mirror directory when they are no longer needed. Cachi2 will want to disable automatic pruning of the offline mirror since we support multiple
+yarn projects in a single request.
+
+To enable the offline mirror, the following settings need to be applied either via .yarnrc or from environment variables:
+
+    yarn-offline-mirror <absolute path to the request output directory>
+    yarn-offline-mirror-pruning false
+
+    YARN_YARN_OFFLINE_MIRROR=<absolute path to the request output directory>
+    YARN_YARN_OFFLINE_MIRROR_PRUNING=false
+
+### Installing Yarn (via Corepack)
+Yarn commands will be executed in order to populate the offline mirror, so Yarn will need to be installed in the Cachi2 image. Corepack is
+already being used to install Yarn in the Cachi2 Yarn v2+ implementation, so a similar approach is suggested.
+
+Corepack installs the latest known-good Yarn v1.x release by default if no other version is specified explicitly. It seems reasonable to always
+use the latest 1.x release for the prefetch, since 1.x is no longer under active development and we never previously allowed users to specify a specific
+Yarn version to process their request in Cachito. We could add more flexibility with a later enhancement if needed.
+
+Corepack could be configured to ignore user project configuration and use the global Yarn 1.x default by specifying the following
+[environment variable](https://github.com/nodejs/corepack#environment-variables):
+
+    COREPACK_ENABLE_PROJECT_SPEC=0
+
+Alternatively we could configure corepack explicitly using the packageManager field in package.json like is done in Cachi2's Yarn v2+ implementation.
```

Comment:
Side note: ^this is an example of where we'd need a decision in the future design doc template to explicitly state which option we went with over a known alternative. It may help in the future if we encounter a problem requiring us to reconsider.